hail/src/main/scala/is/hail/services/DeployConfig.scala

```diff
@@ -122,3 +127,23 @@ class DeployConfig(
     s"${ scheme(baseScheme) }://${ domain(service) }${ basePath(service) }"
   }
 }
+
+class TerraDeployConfig(
+  location: String,
+  defaultNamespace: String,
+  domain: String,
+  subpath: String) extends DeployConfig(location, defaultNamespace, domain) {
+  import DeployConfig._
+
+  override def domain(service: String): String = {
+    this.domain
+  }
+
+  override def basePath(service: String): String = {
+    s"$subpath/$service"
+  }
+
+  override def scheme(baseScheme: String = "http"): String = {
+    "https"
+  }
```

Comment:
Hmm. We always communicate over https? What about driver<->front-end comms, does that not use the local http connection?

Reply:
Currently the only way to communicate with k8s services in Azure Terra is through Azure Relay. I would hope that Azure Relay's smart enough to not have to send that over the internet if you're already in Azure, but I'm not sure. There was a conversation a while back about adding an internal load balancer but the conclusion at the time was they preferred traffic to just go through Relay.